picard/script.py

```diff
@@ -299,6 +300,49 @@ def eval(self, script, context=None, file=None):
         return ScriptParser._cache[key].eval(self)
 
 
+class MultiValue(MutableSequence):
+    def __init__(self, parser, multi, separator):
+        self.parser = parser
+        if isinstance(separator, ScriptExpression):
+            self.separator = separator.eval(self.parser)
+        else:
+            self.separator = separator
+        if (self.separator == MULTI_VALUED_JOINER
+            and len(multi) == 1
+            and isinstance(multi[0], ScriptVariable)):
+            # Convert ScriptExpression containing only a single variable into variable
+            self._multi = self.parser.context.getall(normalize_tagname(multi[0].name))
+        else:
+            # Fall-back to converting to a string and splitting if haystack is an expression
+            # or user has overridden the separator character.
+            evaluated_multi = multi.eval(self.parser)
+            if self.separator:
+                self._multi = evaluated_multi.split(self.separator)
+            else:
+                self._multi = [evaluated_multi]
```

Comment:
This part reads so much clearer then the original code. It is much more clear that there are two cases to consider her.